engine/src/main/java/org/opencds/cqf/cql/engine/elm/execution/RetrieveEvaluator.java

```diff
@@ -22,52 +23,44 @@ protected Object internalEvaluate(Context context) {
         Iterable<Code> codes = null;
         String valueSet = null;
         if (this.getCodes() != null) {
-            if (this.getCodes() instanceof ValueSetRef) {
-                ValueSetRef valueSetRef = (ValueSetRef)this.getCodes();
-                ValueSetDef valueSetDef = context.resolveValueSetRef(valueSetRef.getLibraryName(), valueSetRef.getName());
-                // TODO: Handle value set versions.....
-                valueSet = valueSetDef.getId();
-            }
-            else {
-                Object codesResult = this.getCodes().evaluate(context);
-                if (codesResult instanceof String) {
-                    List<Code> codesList = new ArrayList<>();
-                    codesList.add(new Code().withCode((String)codesResult));
-                    codes = codesList;
-                }
-                else if (codesResult instanceof Code) {
-                    List<Code> codesList = new ArrayList<>();
-                    codesList.add((Code)codesResult);
-                    codes = codesList;
-                }
-                else if (codesResult instanceof Concept) {
-                    List<Code> codesList = new ArrayList<>();
-                    for (Code conceptCode : ((Concept)codesResult).getCodes()) {
-                        codesList.add(conceptCode);
-                    }
-                    codes = codesList;
-                }
-                else {
-                    codes = (Iterable<Code>) codesResult;
+            Object codesResult = this.getCodes().evaluate(context);
+            if (codesResult instanceof ValueSet) {
```

Comment:
It's important to keep the evaluation here as detecting whether there is a ValueSetRef, because 1.4 content will come in this way and the "evaluate" will be interpreted as an expand.